internal/privatemessaging/privatemessaging.go

```diff
@@ -245,7 +245,7 @@ func (pm *privateMessaging) sendData(ctx context.Context, tw *fftypes.TransportW
 		}
 
 		// Send the payload itself
-		err := pm.exchange.SendMessage(ctx, op.ID, node.DX.Peer, payload)
+		err := pm.exchange.SendMessage(ctx, op.ID, node.Profile.GetString("id"), payload)
```

Comment:
This is the concrete example of what I mentioned above - where it is necessary (and enforced) for all DX profiles to have the special field `id` in there. Note this code could have looked up the `verifier` chained off the node (used for the reverse lookup when verifying the off-chain identity of something received), but it seemed easier to just access it directly here rather than taking an extra cached-DB-lookup complexity onboard.